core/src/main/scala/java/util/Currency.scala

```diff
@@ -0,0 +1,76 @@
+package java.util
+
+import locales.LocaleRegistry
+import scala.collection.{Map => SMap, Set => SSet}
+import locales.cldr.CurrencyDataInfo
+import locales.cldr.data.currencydata
+
+object Currency {
+  private val countryCodeToCurrencyCodeMap: SMap[String, String] = currencydata.regions.map{ r =>
+    r.countryCode -> r.currencies.find{ _.to.isEmpty }.map{ _.currencyCode }.get
+  }.toMap
+
+  private val all: SSet[Currency] = currencydata.infos.map{ info: CurrencyDataInfo =>
+    Currency(info.currencyCode, info.numericCode, info.digits, None)
+  }.toSet
+
+  private val currencyCodeMap: SMap[String, Currency] = all.toSeq.groupBy{ _.getCurrencyCode }.map{
+    case (currencyCode: String, matches: Seq[Currency]) => currencyCode -> matches.head
+  }
+
+  def getAvailableCurrencies(): SSet[Currency] = all
+
+  // NullPointerException - if locale or its country code is null
+  // IllegalArgumentException - if the country of the given locale is not a supported ISO 3166 country code.
+  def getInstance(locale: Locale): Currency = {
+    if (locale.getCountry == null || locale.getCountry.isEmpty) throw new NullPointerException
```

Comment:
Does the JVM throw an NPE in this case? I mean at runtime rather than what the javadoc says